src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -242,9 +241,8 @@ private void OnRequestPaste()
         /// <param name="workspace">workspace view</param>
         private void PasteNodeAtTheCenter(WorkspaceView workspace)
         {
-            var centerX = (workspace.ActualWidth / 2 - workspace.ViewModel.Model.X) / workspace.ViewModel.Zoom;
-            var centerY = (workspace.ActualHeight / 2 - workspace.ViewModel.Model.Y) / workspace.ViewModel.Zoom;
-            dynamoViewModel.Model.Paste(new Point2D(centerX, centerY));
+            var centerPoint = workspace.GetCenterPoint().AsDynamoType();
+            dynamoViewModel.Model.Paste(centerPoint);
```

Comment:
the previous implementation gives incorrect `centerX`, `centerY` if the workspace is just created and its position hasn't been changed yet